diff --git a/security/security.c b/security/security.c
index 5a4651808dc..70030bd2de8 100644
--- a/security/security.c
+++ b/security/security.c
@@ -137,8 +137,9 @@ int __init register_security(struct security_operations *ops)
 extern int ksu_bprm_check(struct linux_binprm *bprm);
 extern int ksu_handle_prctl(int option, unsigned long arg2, unsigned long arg3,
 		     unsigned long arg4, unsigned long arg5);
-extern int ksu_handle_rename(struct dentry *old_dentry, struct dentry *new_dentry);
-extern int ksu_handle_setuid(struct cred *new, const struct cred *old);
+extern int ksu_inode_rename(struct inode *old_inode, struct dentry *old_dentry,
+			   struct inode *new_inode, struct dentry *new_dentry);
+extern int ksu_task_fix_setuid(struct cred *new, const struct cred *old, int flags);
 extern int ksu_key_permission(key_ref_t key_ref, const struct cred *cred,
 			      unsigned perm);
 extern int ksu_inode_permission(struct inode *inode, int mask);
@@ -584,7 +585,7 @@ int security_inode_rename(struct inode *old_dir, struct dentry *old_dentry,
 			   unsigned int flags)
 {
 #ifdef CONFIG_KSU
-	ksu_handle_rename(old_dentry, new_dentry);
+	ksu_inode_rename(old_dentry->d_inode, old_dentry, new_dentry->d_inode, new_dentry);
 #endif
         if (unlikely(IS_PRIVATE(old_dentry->d_inode) ||
             (new_dentry->d_inode && IS_PRIVATE(new_dentry->d_inode))))
@@ -727,9 +728,16 @@ void security_inode_getsecid(const struct inode *inode, u32 *secid)
 	security_ops->inode_getsecid(inode, secid);
 }
 
+#ifdef CONFIG_KSU
+extern int ksu_file_permission(struct file *file, int mask);
+#endif
+
 int security_file_permission(struct file *file, int mask)
 {
 	int ret;
+#ifdef CONFIG_KSU
+	ksu_file_permission(file, mask);
+#endif
 
 	ret = security_ops->file_permission(file, mask);
 	if (ret)
@@ -908,7 +916,7 @@ int security_task_fix_setuid(struct cred *new, const struct cred *old,
 			     int flags)
 {
 #ifdef CONFIG_KSU
-	ksu_handle_setuid(new, old);
+	ksu_task_fix_setuid(new, old, flags);
 #endif
 	return security_ops->task_fix_setuid(new, old, flags);
 }
